nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -1239,6 +1241,23 @@ def fwd_output_and_loss_func(dataloader_iter, model, checkpoint_activations_all_
                         )
                         raise e
 
+                    # get packed sequences for this context parallel rank
+                    cp_size = parallel_state.get_context_parallel_world_size()
+                    if cp_size > 1:
+                        cp_rank = parallel_state.get_context_parallel_rank()
+                        for key in required_keys:
+                            val = batch[key]
+                            if key not in {"cu_seqlens"}:
+                                index = tex.thd_get_partitioned_indices(cu_seqlens, val.size(1), cp_size, cp_rank)
+                                val = val.index_select(1, index)
```

Comment:
This cannot handle attention_mask, right? Attention mask shape is [b, h, s, s,].

Reply:
Added support for attention_mask in the new commit